packages/flutter/lib/src/rendering/mouse_tracking.dart

```diff
@@ -418,16 +421,17 @@ class BaseMouseTracker extends ChangeNotifier {
       || lastEvent.position != event.position;
   }
 
-  // Find the annotations that is hovered by the device of the `state`.
+  // Find the annotations that is hovered by the device of the `state`, and
+  // their respective global transform matrices.
   //
-  // If the device is not connected, an empty set is returned without calling
-  // `annotationFinder`.
-  LinkedHashSet<MouseTrackerAnnotation> _findAnnotations(_MouseState state) {
+  // If the device is not connected or not a mouse, an empty map is returned
+  // without calling `annotationFinder`.
+  LinkedHashMap<MouseTrackerAnnotation, Matrix4> _findAnnotations(_MouseState state) {
     final Offset globalPosition = state.latestEvent.position;
     final int device = state.device;
-    return (_mouseStates.containsKey(device))
-      ? LinkedHashSet<MouseTrackerAnnotation>.from(annotationFinder(globalPosition))
-      : <MouseTrackerAnnotation>{} as LinkedHashSet<MouseTrackerAnnotation>;
+    if (!_mouseStates.containsKey(device))
+      return <MouseTrackerAnnotation, Matrix4>{} as LinkedHashMap<MouseTrackerAnnotation, Matrix4>;
```

Comment:
Why not just do this?:
```suggestion
      return LinkedHashMap<MouseTrackerAnnotation, Matrix4>();
```

Reply:
The linter.
With `LinkedHashMap<MouseTrackerAnnotation, Matrix4>()` the linter reports "Use literal when possible". 
With `<MouseTrackerAnnotation, Matrix4>{}` the compiler reports a type mismatch.
This is the last I can do...

Reply:
Well, that is pretty confusing.  I guess it's because the real type of `Map` is `_InternalLinkedHashMap` and not `LinkedHashMap`. It's probably no less efficient, since I'll bet `LinkedHashMap` is implemented as an `_InternalLinkedHashMap` internally, but ewww...

Reply:
I brought it up in https://github.com/dart-lang/linter/issues/1649#issuecomment-656929040. Hopefully it raises some attention :/